cores/esp32/esp32-hal-bt.c

```diff
@@ -20,30 +20,44 @@ bool btInUse(){ return true; }
 
 #include "esp_bt.h"
 
+#ifndef BT_MODE
 #ifdef CONFIG_BTDM_CONTROLLER_MODE_BTDM
 #define BT_MODE ESP_BT_MODE_BTDM
 #elif defined(CONFIG_BTDM_CONTROLLER_MODE_BR_EDR_ONLY)
 #define BT_MODE ESP_BT_MODE_CLASSIC_BT
 #else
 #define BT_MODE ESP_BT_MODE_BLE
 #endif
+#endif
 
 bool btStarted(){
     return (esp_bt_controller_get_status() == ESP_BT_CONTROLLER_STATUS_ENABLED);
 }
 
 bool btStart(){
     esp_bt_controller_config_t cfg = BT_CONTROLLER_INIT_CONFIG_DEFAULT();
+#if BT_MODE == ESP_BT_MODE_CLASSIC_BT
+    // esp_bt_controller_enable(MODE) This mode must be equal as the mode in “cfg” of esp_bt_controller_init().
+    esp_bt_controller_mem_release(ESP_BT_MODE_BLE);
+#elif BT_MODE == ESP_BT_MODE_BLE
+    esp_bt_controller_mem_release(ESP_BT_MODE_CLASSIC_BT);
+#endif
+    cfg.mode=BT_MODE;
```

Comment:
`cfg.mode` is only available on ESP32. You should have kept the whole block filtered for ESP32 target. Also as far as I remember, doing mem release of the classic BT will also break BLE. If BT/BLE is not used, that memory is already cleared when Arduino is starting up. So keep just `esp_bt_controller_mem_release(ESP_BT_MODE_BLE);`

Reply:
sorry, you were too fast.
1. esp_bt_controller_mem_release(ESP_BT_MODE_CLASSIC_BT) crashes the esp32 without an error message when using BLE
2. unexpectedly #if BT_MODE == ESP_BT_MODE_CLASSIC_BT doesn't work. The preprocessor only likes integer values, I will find a solution. Thinking of:
 ```
    cfg.mode=BT_MODE;
    if(cfg.mode == ESP_BT_MODE_CLASSIC_BT) {
      esp_bt_controller_mem_release(ESP_BT_MODE_BLE);
    }
```
tomorrow....

Reply:
```cpp
#if CONFIG_IDF_TARGET_ESP32
cfg.mode=BT_MODE;
if(cfg.mode == ESP_BT_MODE_CLASSIC_BT) {
  esp_bt_controller_mem_release(ESP_BT_MODE_BLE);
}
#endif
```

Looks like a good plan 👍

Reply:
@me-no-dev Is it not possible to clear memory used by BT_CLASSIC if I'm using only BLE? In the case of Rainmaker, we often run into low memory since both Classic BT and BLE are on during WiFi provisioning. Tagging @shahpiyushv.

Reply:
@sanketwadekar as far as I remember and it seems that @ferbar confirms it above, it is not possible to release classic memory even if you are using just BLE